File: RepoTransBench/target_projects/Java/C#/Meituan-Dianping_walle/tests/original/ApkUtilTest.cs
using System;
using System.IO;
using Xunit;
using MeituanDianpingWalle;

namespace MeituanDianpingWalle.Tests
{
    public class ApkUtilTest
    {
        [Fact]
        public void TestGetApkSigningBlock_NullInput()
        {
            Assert.Throws<ArgumentNullException>(() => ApkUtil.GetApkSigningBlock(null));
        }

        [Fact]
        public void TestGetMapIdValue_EmptyArray()
        {
            Assert.Null(ApkUtil.GetMapIdValue(new ApkUtil.Pair[0], 0L));
        }

        [Fact]
        public void TestFindApkSignatureSchemeV2BlockId()
        {
            var dummy = new FileInfo("non-existent.apk");
            var ex = Record.Exception(() => ApkUtil.FindApkSignatureSchemeV2BlockId(dummy));
            Assert.NotNull(ex);
            Assert.True(ex is IOException);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/Meituan-Dianping_walle/tests/original/apkutil_test.go
package original

import (
	"errors"
	"os"
	"testing"
)

type ApkUtil struct{}

type Pair struct {
	First  interface{}
	Second interface{}
}

func (a *ApkUtil) GetApkSigningBlock(f *os.File) error {
	if f == nil {
		return errors.New("nil input")
	}
	return nil
}

func (a *ApkUtil) GetMapIdValue(arr []Pair, id int64) interface{} {
	// Return nil if array is empty
	if len(arr) == 0 {
		return nil
	}
	for _, p := range arr {
		if p.First == id {
			return p.Second
		}
	}
	return nil
}

func (a *ApkUtil) FindApkSignatureSchemeV2BlockId(f *os.File) error {
	if f == nil {
		return errors.New("nil input")
	}
	_, err := os.Stat(f.Name())
	if err != nil {
		return err
	}
	return nil
}

func TestApkUtil_GetApkSigningBlock_NullInput(t *testing.T) {
	apkUtil := &ApkUtil{}
	defer func() {
		if r := recover(); r == nil {
			t.Errorf("Should panic for nil input")
		}
	}()
	apkUtil.GetApkSigningBlock(nil)
}

func TestApkUtil_GetMapIdValue_EmptyArray(t *testing.T) {
	apkUtil := &ApkUtil{}
	if apkUtil.GetMapIdValue([]Pair{}, 0) != nil {
		t.Fatalf("should return nil for empty input")
	}
}

func TestApkUtil_FindApkSignatureSchemeV2BlockId(t *testing.T) {
	apkUtil := &ApkUtil{}
	dummyFile := "non-existent.apk"
	file, err := os.Open(dummyFile)
	// file won't exist, simulate error
	if err == nil {
		file.Close()
	}
	err = apkUtil.FindApkSignatureSchemeV2BlockId(&os.File{Name: func() string { return dummyFile }()})
	if err == nil {
		t.Fatalf("Should throw error for non-existent file")
	}
}